skfda/datasets/_samples_generators.py

```diff
@@ -135,104 +255,296 @@ def euler_maruyama(  # noqa: WPS210
+    include it.
+
+    Args:
+        initial_condition: Initial condition of the SDE. It can have one of
+            three formats: An starting initial value from which to
+            calculate *n_samples* trajectories. An array of initial values.
+            For each starting point a trajectory will be calculated. A
+            function that generates random numbers or vectors. It should
+            have two parameters called size and random_state and it should
+            return an array.
+        drift: Drift term (:math:`F(t,\mathbf{X})` in the equation).
+        diffusion: Diffusion term (:math:`G(t,\mathbf{X})` in the
+            equation). The diffusion term should depend on the variable
+            :math:`\mathbf{X}`.
+        diffusion_derivative: Derivate of the diffusion term
+            (:math:`G_\mathbf{X}(t, \mathbf{X})`). The return of this function
+            should
+        n_L0_discretization_points: number of discretization points to
+            approximate the double Itô integral :math:`I_{(j_1,j_2)}`. Only
+            use when the process is multidimensional.
+        n_grid_points: The total number of points of evaluation.
+        n_samples: Number of trajectories integrated.
+        start: Starting time of the trajectories.
+        stop: Ending time of the trajectories.
+        diffusion_matricial_term: True if the diffusion coefficient is a
+            matrix.
+        random_state: Random state.
+
+    Returns:
+        :class:`FDataGrid` object comprising all the trajectories.
+
+    See also:
+        :func:`euler_maruyama`: more simple numerical integrator of SDEs
+        with lower order of convergence.
+
+    Examples:
+        Example of the use of milstein for a 1-d Geometric Brownian motion
+        that has the equation:
+
+        ..  math:
+
+            dX(t) = \mu X(t) dt + \sigma X(t) dW(t)
+
+        >>> sigma = 1
+        >>> mu = 2
+        >>> def gbm_drift(t: float, x: np.ndarray) -> np.ndarray:
+        ...     return mu * x
+        >>> def gbm_diffusion(t: float, x: np.ndarray) -> np.ndarray:
+        ...     return sigma * x
+        >>> def gbm_diff_derivative(t: float, x: np.ndarray) -> np.ndarray:
+        ...     return sigma * np.ones_like(x)[: :, np.newaxis]
+        >>> X_0 = 1
+        >>> n_L0_discretization_points = 5,
+        >>>
+        >>> trajectories = milstein(
+        ...     X_0,
+        ...     gbm_drift,
+        ...     gbm_diffusion,
+        ...     gbm_diff_derivative,
+        ...     n_L0_discretization_points,
+        ...     n_samples=10,
+        ... )
+
+    References:
+        Paromita Banerjee, Brynjarsdottir Jenny, Gurarie David, and
+        Albert Jeffrey. Numerical Methods for Stochastic Differential Equations
+        and Postintervention in Structural Equation Models. PhD thesis, USA,
+        2020. AAI29605510
+    """
+    random_state = validate_random_state(random_state)
```

Comment:
I do not have the bandwidth now to check for code correctness in depth. I will assume that Alberto has reviewed the code and tests with you thoroughly. If that is not true, please let me now to recheck it later before merging.